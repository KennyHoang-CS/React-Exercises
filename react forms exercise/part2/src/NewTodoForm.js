import React, { useState } from 'react';

const NewTodoForm = ({ addTask }) => {
    const INITIAL_STATE = {
        task: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleChange = (e) => {
        const { name, value } = e.target; 
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        addTask({ ...formData });
        setFormData(INITIAL_STATE);
    }
    return (
        <form onSubmit={ handleSubmit }>
            <label htmlFor="task">New Task</label>
            <input
                type="text"
                id="task"
                name="task"
                placeholder="new task"
                value={ formData.task }
                onChange={handleChange}
            />
        <button>Add new task!</button>
        </form>
    )
}

export default NewTodoForm;