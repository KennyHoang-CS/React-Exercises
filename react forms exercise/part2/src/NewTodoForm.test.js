import { render, screen } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it('NewTodoForm should render', () => {
  render(<NewTodoForm/>);
});

it('NewTodoForm matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm/>);
  expect(asFragment()).toMatchSnapshot();
});
