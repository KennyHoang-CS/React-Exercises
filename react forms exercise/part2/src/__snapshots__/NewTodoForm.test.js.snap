// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      New Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="new task"
      type="text"
      value=""
    />
    <button>
      Add new task!
    </button>
  </form>
</DocumentFragment>
`;
